aiida/manage/tests/unittest_classes.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+Test classes and test runners for testing AiiDA plugins with unittest.
+"""
+
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import print_function
+import unittest
+
+from aiida.manage.manager import get_manager
+from . import _GLOBAL_TEST_MANAGER, test_manager, get_test_backend
+
+__all__ = ('PluginTestCase', 'TestRunner')
+
+
+class PluginTestCase(unittest.TestCase):
```

Comment:
Do we want to keep supporting `unittest`? I thought we discussed deprecating this to be removed in `2.0.0` and directing people to use `pytest` instead

Reply:
Good point - the docs already state that `pytest`  is the preferred way.

However, we can add a deprecation warning here as well.

Reply:
If we are not going to support it, which I don't think we should, we should add the deprecation warning

Reply:
Well, the `PluginTestCase` has been in AiiDA since some time and people could rely on it.
This goes back to our promise not to break the API after 1.0.0b2.

Reply:
I am not saying that we remove it now or even in `1.*` but we can deprecate it and remove it in `2.0.0` no? Unless you think it makes sense to keep it and keep supporting it of course

Reply:
Oops, sorry, I misread your comment.
I did add the deprecation warning in the TestRunner, not at the module level (because I wanted to avoid deprecation warnings popping up accidentally).

Let me know whether you think there is a better place to put it.
I'm even ok with moving it at the module level if you think this is better.

Reply:
Ah, I missed that one, then it is fine :+1: